### DISCUSSION TITLE
A issue about folding 3D four-bar linkage

### DISCUSSION THREAD

**Discussion by KaiXiao55** [edemaine/fold#33] [Q&A] [answered]
Dear authors,

Thanks for this great origami simulator. I really enjoy playing with it.

However, it looks like I encountered a small problem to fold a three-dimensional four-bar linkage.

Specifically, after importing my four-bar linkage FOLD file, it exhibits the correct shape of linkage, but the hinges don't deform when I was changing the folding ratio (See attached figures). I wonder why it does not fold, is that because of something wrong with my FOLD file (attached)?

I will really appreciate it if you reply.

Thanks! 

Kai Xiao.

[3D_fourBar.txt](https://github.com/edemaine/fold/files/5975770/3D_fourBar.txt)

![3D four-bar linkage_Flat](https://user-images.githubusercontent.com/75075763/107851414-a4ffef80-6e44-11eb-9a49-391690cd574f.png)
![3D four-bar linkage_Folded](https://user-images.githubusercontent.com/75075763/107851415-a7fae000-6e44-11eb-83dc-ba78255d0596.png)

**Comment by edemaine** [comment, marked as answer]
I think you meant to post this under [Origami Simulator](https://github.com/amandaghassaei/OrigamiSimulator)?  Anyway, I think you need to add an `edges_foldAngle` field to specify the target fold angles for your model (probably some to zero and some to 180?); otherwise, Origami Simulator doesn't know what you want to fold it into.

**Reply by KaiXiao55** [reply]
Really appreciate your reply!  Now my FOLD file works.
